catalog/github/webhook.js

```diff
@@ -53,12 +53,29 @@ function main(params) {
 
     request(options, function(error, response, body) {
       if (error) {
-        whisk.error();
+        whisk.error({
+          response: response,
+          error: error,
+          body: body
+        });
+      } else {
+        console.log("Status code: " + response.statusCode);
+
+        if(response.statusCode >= 400) {
+          console.log("Response from Github: " + body);
+          whisk.error({
+            statusCode: response.statusCode,
+            response: body
+          });
+        } else {
+          whisk.done({response: body});
+        }
       }
-      console.log("Status code: " + response.statusCode);
-      whisk.done({response: body});
     });
+
     return whisk.async();
```

Comment:
I feel bad for asking, given how long I've left this open, but could we use a `Promise`? I think we're already operating internally with the mental model that `.async()` and friends are on their way to being marked deprecated?